### ISSUE TITLE
Support for Fresh Air Appliance (some type of air conditioner) devices (0xCE)

### ISSUE THREAD

**Issue by kovapatrik** [kovapatrik/homebridge-midea-platform#118]
[ce_lua.zip](https://github.com/user-attachments/files/18125606/ce_lua.zip)

**Comment by feedericob** [comment]
Thanks

**Comment by kovapatrik** [comment]
can you send me some screenshots from the app how the interface looks like for this device? I just want to see what i should implement, what accessories to add, etc...

**Comment by feedericob** [comment]
![IMG_3495](https://github.com/user-attachments/assets/1cf7c490-1c0e-46ab-ba57-21a7d3b18a41)
![IMG_3496](https://github.com/user-attachments/assets/cc9848e7-d170-48b3-8b1e-66619149c2ba)
 Thanks

**Comment by feedericob** [comment]
Hi,
probably may you check the external temperature...


Best

**Comment by kovapatrik** [comment]
I will check it, sure! I just don't know when I will have time to do this.

**Comment by feedericob** [comment]
No problem

**Comment by feedericob** [comment]
Hi,
any news?

Best

**Comment by kovapatrik** [comment]
Sorry, I still did not have time/motivation to fix these issues. I will try to do it in the next weeks.

**Comment by kovapatrik** [comment]
I started to implement this now. Does your device report PM2.5 or CO2 measurements?

**Comment by feedericob** [comment]
Hi, no there is no CO2 measurement. The external temperature appears on the screen of the machine but actually does not appear in the app.

No, the schedule is not needed.

One detail, in the silent there are 3 levels. OFF, Silent, Supersilent.

![Image](https://github.com/user-attachments/assets/4028e1a6-63d3-4ea7-99f1-af2254a38d1e)

**Comment by kovapatrik** [comment]
Thank you!

What does the `AutoSet` mode means? How is it different from `Auto`?

**Comment by feedericob** [comment]
Basically the necessary states are OFF, Cool, Heat.

Being a machine that automatically modulates its operation, it makes no sense to put it in automatic.

From display you can also put Auto. Autoset on the machine does not exist.

The app also warns you of any machine errors. I don't know if Apple has an alert, which can be viewed in the accessory. Or an error sensor.

**Comment by kovapatrik** [comment]
Yes, I get the error codes and it's possible to set the device as errored in Apple Home, or I can create a sensor which will show the errors. I just don't know what the codes mean, so after I made the basic implementation, I would need you to reverse engineer this. Can you manually trigger the errors you need to be handled?

**Comment by feedericob** [comment]
Some errors block the machine, others warn you if the filters need to be changed. If you don't receive notifications, you won't notice if there are. The machine emits a sound but you don't always hear it if it's installed in the ceiling. I think it's already great that a notification like crtitical error comes out, or a sensor that can then activate a notification.

I'll attach all the errors anyway

![Image](https://github.com/user-attachments/assets/8c288089-32ba-4940-a175-cc5c01d494fc)
![Image](https://github.com/user-attachments/assets/239e9c98-8367-49f9-97f2-991e78f8cc2e)

**Comment by kovapatrik** [comment]
It's not possible to create push notifications from the Home app (or from Homebridge), but there are apps for this: https://linkdhome.com/articles/custom-homekit-notifications

**Comment by feedericob** [comment]
A critical alarm like that of security systems? Otherwise ok a sensor, which if it closes anyway appears.

**Comment by kovapatrik** [comment]
Wait, do you have devices in Home which can send you notifications?

**Comment by feedericob** [comment]
Of course, the alarm systems send notifications if there is intrusion. Then the shutters and gates have the possibility to activate the notification when they open or close. The sensors too.

![Image](https://github.com/user-attachments/assets/bdeb4c77-f60c-457a-9d25-553a9ed9ebff)
![Image](https://github.com/user-attachments/assets/f0e86ca9-ebd7-46b5-8c69-3cd77c486c3d)
![Image](https://github.com/user-attachments/assets/7f20acf5-2fce-41b7-bd82-0ac385338e0e)
![Image](https://github.com/user-attachments/assets/ae8ce590-3f54-4cf8-8d94-264f264be4d0)

**Comment by kovapatrik** [comment]
Hmm, I think this depends on the type of the device. For example my air conditioner does not have a `Status and Notifications` menu.

**Comment by feedericob** [comment]
No, of course not everyone has it. However, if you show up an active/deactivated sensor, you can be notified

**Comment by feedericob** [comment]
Hi, devices can now be added. However, I can't choose between air conditioning and fresh ari appliance and other services.

[homebridge.log.txt](https://github.com/user-attachments/files/19541350/homebridge.log.txt)

<img width="781" alt="Image" src="https://github.com/user-attachments/assets/bb0587b1-aca1-4c09-8685-3935340dd5a7" />

<img width="826" alt="Image" src="https://github.com/user-attachments/assets/314e729d-eb42-41ae-8cfe-2a89d71cd5e8" />

**Comment by kovapatrik** [comment]
Do you mean you don't know which one to select? Your device's model is called `Fresh Air Appliance`. Usually, the default value is valid.

How can I make it more user friendly which device is which, do you have a suggestion? The `Fresh Air Appliance` should be also visible in the discovery table btw.

**Comment by kovapatrik** [comment]
It would be also nice to see if handling the error code works. Is there a basic error which you can trigger manually? Like removing the filter. (I don't know what your device does, so I don't know if it has a filter or not, but I think you got what I want to do)

**Comment by kovapatrik** [comment]
Any news on this?

**Comment by feedericob** [comment]
I haven't been able to check it yet, I should be able to do it this week. Thanks

**Comment by feedericob** [comment]
Hi, 
I updated the app to the latest version. I attach some screenshots.

In the accessory settings I see the current and set temperature correctly, but in the accessory I can choose between humidifier and dehumidifier, this machine instead must have the settings of a classic thermostat.

Currently it only works on and off.

I tried to change the device type from the plugin settings but it doesn't take it.


![Image](https://github.com/user-attachments/assets/471b40e1-0e4d-43ce-bcf2-a36cc98e40a1)
![Image](https://github.com/user-attachments/assets/0aa59f7e-0509-4b25-a80c-d230d07a8483)
![Image](https://github.com/user-attachments/assets/c8758a3d-0dd6-4a2e-967a-9560560b56ab)

Classic thermostat

![Image](https://github.com/user-attachments/assets/aac940a5-255e-45ef-968b-a9b884b58913)

Thanks
